Cargo.toml

```diff
@@ -2,6 +2,7 @@
 resolver = "2"
 
 members = [
+  "bin",
```

Comment:
Hey @johnletey, thanks for the PR! I've reviewed the changes, and most of them look great. The one part I'd like to discuss is the separation into a `bin/` crate. Currently, with `app/` containing both `lib.rs` and `main.rs`, we already have the library functionality available, and this is a pretty [standard Rust pattern](https://doc.rust-lang.org/stable/book/ch07-01-packages-and-crates.html#:~:text=Cargo%20follows%20a,library%20or%20binary.). The binary and library coexist well in the same crate. I'm curious about your reasoning for the additional `bin/` separation - is there a particular reason it is important for your use case?

Reply:
@bastienfaivre If I'm understanding you, you want to simply add a `lib.rs` and keep everything else the same? We're fine with doing that, but the only thing existing right now was the `main.go`

Reply:
I mean, keeping all the changes you've made, except removing the `bin` member in the main `Cargo.toml` and the corresponding `bin/Cargo.toml` directory/file.
Let me know if, by doing so, you encounter issues when using Emerald as a lib.

Reply:
I've made the requested changes!